codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -572,46 +573,77 @@ private class ServerHttpProtocolImplGenerator(
+                    if (segment.isLabel) { "m$index" } else { "_" }
                 }
-                pattern.append(")")
-            } else {
-                pattern.append(it.content)
+                .joinToString(separator = ",")
+            val nomParser = segments
+                .map { segment ->
+                    if (segment.isGreedyLabel) {
+                        "#{Nom}::combinator::rest"
+                    } else if (segment.isLabel) {
+                        """#{Nom}::branch::alt((#{Nom}::bytes::complete::take_until1("/"), #{Nom}::combinator::rest))"""
+                    } else {
+                        """#{Nom}::bytes::complete::tag("${segment.content}")"""
+                    }
+                }
+                .joinToString(
+                    prefix = "#{Nom}::sequence::tuple::<_, _, #{Nom}::error::Error<&str>, _>((",
+                    postfix = "))",
+                    transform = { parser ->
+                        """
+                        #{Nom}::sequence::preceded(#{Nom}::bytes::complete::tag("/"),  $parser)
+                        """.trimIndent()
+                    }
+                )
+            rustTemplate("let input_string = request.uri().path();")
+            if (restAfterGreedyLabel.isNotEmpty()) {
+                rustTemplate(
+                    """
+                    if ! input_string.ends_with(${restAfterGreedyLabel.dq()}) {
+                        return std::result::Result::Err(#{SmithyRejection}::MissingQueryString(aws_smithy_http_server::rejection::MissingQueryString));
+                    }
+                    let input_string = input_string[..(input_string.len() - ${restAfterGreedyLabel.dq()}.len())].into();
+                    """.trimIndent(),
+                    *codegenScope
+                )
             }
-        }
-        with(writer) {
             rustTemplate(
                 """
-                #{LazyStatic}::lazy_static! {
-                    static ref RE: #{Regex}::Regex = #{Regex}::Regex::new("$pattern").unwrap();
-                }
+                let (_input_string, ($labeledNames)) = $nomParser(input_string)?;
                 """.trimIndent(),
-                *codegenScope,
+                *codegenScope
             )
-            rustBlock("if let Some(captures) = RE.captures(request.uri().path())") {
-                pathBindings.forEach {
-                    val deserializer = generateParsePercentEncodedStrFn(it)
-                    rustTemplate(
-                        """
-                        if let Some(m) = captures.name("${it.locationName}") {
-                            input = input.${it.member.setterName()}(
-                                #{deserializer}(m.as_str())?
+            segments
+                .forEachIndexed { index, segment ->
+                    val binding = pathBindings.find { it.memberName == segment.content }
+                    if (binding != null && segment.isLabel) {
```

Comment:
If `segment.isLabel` is `true`, then `binding` must be non-null, right? Because `pathBindings` contains all the `@httpLabel` bindings.

Reply:
In some cases, such as `/foo/{foo}`, `foo` is found as binding too because I look for the name